server/src/main/java/org/elasticsearch/cluster/routing/IndexRouting.java

```diff
@@ -295,22 +292,20 @@ public boolean matchesField(String fieldName) {
         }
 
         @Override
-        public void process(IndexRequest indexRequest) {}
+        public void postProcess(IndexRequest indexRequest) {
```

Comment:
I don't completely follow why `IdAndRoutingOnly` only implements preProcess(...) and `ExtractFromSource` only implements `postProcess(...)`. Can you explain this?

Reply:
In `IdAndRoutingOnly` the id needs to get populated before routing, as it's used to find the shard to route to. In `ExtractFromSource` we first need to evaluate the routing hash (during routing), then use it to build the id. The latter is the case for both TSDB and LogsDB with routing on sort fields.

Reply:
Ah, right makes sense. Maybe add this as a comment?